rrrspec-server/lib/rrrspec/server/worker_runner.rb

```diff
@@ -178,9 +178,14 @@ def cleaning_process(logger, taskset, cancel_watcher_pid, pid_to_slave_number)
 
       def work
         @worker.update_current_taskset(nil)
-        taskset = @worker.dequeue_taskset
-        worker_log = WorkerLog.create(@worker, taskset)
-        logger = TimedLogger.new(worker_log)
+
+        while true do
+          taskset = @worker.dequeue_taskset
+          worker_log = WorkerLog.create(@worker, taskset)
+          logger = TimedLogger.new(worker_log)
+
+          break if taskset.status == 'running'
```

Comment:
Can't use check.call because it would return from the work method. 
The fix relies on not notifying the master if the worker has tasksets in the queue. After returning from work method a notification to master is sent.